### ISSUE TITLE
[HOLD for payment 2024-02-14] [$1000]  Whole thread disappears if the thread header is flagged as an Assault or Harassment

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#22383]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1, Send a message from user A to user B
2. From the tab where user B is signed-in, reply to the message in a thread
3. Flag the first message in the thread - the message you replied to - as Harassment or Assault
4. Click on the thread header
## Expected Result: 
Deleted placeholder should be displayed replacing the flagged message. And the whole thread shouldn't be deleted.
## Actual Result: 
The whole thread disappears

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.37-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/ee0370f1-78c0-4261-bbaa-c7b6e3536d15

https://github.com/Expensify/App/assets/43996225/c6731da1-9571-464c-9847-e16ecc6b1b2a

**Expensify/Expensify Issue URL:**
**Issue reported by:** @natnael-guchima
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1688582784260459

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~016effa328d9476bd5</li>
        <li>Upwork Job ID: 1680945434406293504</li>
        <li>Last Price Increase: 2023-07-24</li>
<li>Automatic offers: </li>
<ul>
<li>situchan | Reviewer | 26810740</li>
<li>tienifr | Contributor | 26810743</li>
<li>natnael-guchima | Reporter | 26810745</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @johncschuster</details>

**Comment by jeet-dhandha** [comment]
Mostly due to this issue https://github.com/Expensify/App/issues/22311

**Comment by s77rt** [comment]
Indeed this not a bug. It's done on purpose.

**Comment by johncschuster** [comment]
Thanks for calling this out! I'm checking the design of this feature to ensure this is really intended. I suspect it is, but I'll have a definitive answer shortly!

**Comment by johncschuster** [comment]
Ok! It sounds like this behavior might not be intended. (See [Slack thread](https://expensify.slack.com/archives/C049HHMV9SM/p1688582784260459) for discussion)

It seems we're discussing some of the behavior we'd like to see in the LHN when a thread is removed for being flagged ([GH](https://github.com/Expensify/App/issues/21890)), but I don't think that should be a blocker for this issue.

**Comment by johncschuster** [comment]
Ok! We're going to treat this like a new issue (rather than a regression).

@dangrous, I'm going to loop you into this one so we can both loosely follow the proposals and make sure they fit into the direction we decided on.

**Comment by Talha345** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

When a parent thread message is flagged as assault or harassment, the entire thread disappears from the LHN and a 404 appears when clicked on the header icon.

### What is the root cause of that problem?

PR https://github.com/Expensify/App/pull/21854   implemented this functionality so the changes in this PR are the root cause of this problem.

### What changes do you think we should make in order to solve the problem?

1. The changes in https://github.com/Expensify/App/pull/21854 should be reverted.
2. The following if clause should be modified as follows:

https://github.com/Expensify/App/blob/9004c3968fcde7f46a473db871b4dd6ab2f3effc/src/libs/ReportUtils.js#L1129-L1134

```
const decision = lodashGet(parentReportAction, 'message[0].moderationDecisions[0].decision');
if (
            [CONST.MODERATION.MODERATOR_DECISION_PENDING_HIDE,
                CONST.MODERATION.MODERATOR_DECISION_HIDDEN,
                CONST.MODERATION.MODERATOR_DECISION_PENDING_REMOVE].includes(decision)
        ) {
            return Localize.translateLocal('parentReportAction.hiddenMessage');
        }
```

This will just display [Hidden message] in the LHN, thread header and sidebar.

If it is decided to display [Deleted message] instead ,then the above code can be modified as follows:

```
       if (
            [CONST.MODERATION.MODERATOR_DECISION_PENDING_HIDE,
                CONST.MODERATION.MODERATOR_DECISION_HIDDEN].includes(decision)
        ) {
            return Localize.translateLocal('parentReportAction.hiddenMessage');
        }
        if (
            decision === CONST.MODERATION.MODERATOR_DECISION_PENDING_REMOVE) {
            return Localize.translateLocal('parentReportAction.deletedMessage');
        }
```


### What alternative solutions did you explore? (Optional)

N/A



### Result:

![image](https://github.com/Expensify/App/assets/36226639/20097a66-e872-43f8-b005-3bbec59ebaf0)

**Comment by johncschuster** [comment]
@aimane-chnaif how are these proposals looking to you? Are they sufficient, or should we keep looking?

**Comment by Talha345** [comment]
@johncschuster @aimane-chnaif Looks like melvin is not happy! 😅

**Comment by johncschuster** [comment]
Thanks, @aimane-chnaif!

**Comment by johncschuster** [comment]
@aimane-chnaif did you get a chance to review the above proposals? Are they sufficient, or should we increase the bounty?

**Comment by Talha345** [comment]
> @aimane-chnaif did you get a chance to review the above proposals? Are they sufficient, or should we increase the bounty?

I think the proposals haven't been reviewed yet because in either case some feedback is appreciated from the C+ member on the existing proposals.

**Comment by Talha345** [comment]
> w don't post code diff in pr

Hi @aimane-chnaif, can you please clarify what do you mean by don't post code diff in proposal because in my [proposal](https://github.com/Expensify/App/issues/22383#issuecomment-1638723258) , I have just pointed the code which is causing the issue and then posted some modifications to it which i recommend as my solution.

**Comment by Talha345** [comment]
@aimane-chnaif @johncschuster The proposals have still not been reviewed.

**Comment by Talha345** [comment]
> @Talha345 We don't need to fully revert https://github.com/Expensify/App/pull/21854

@aimane-chnaif Yes I know only the functionality that caused this.

**Comment by Talha345** [comment]
> After applying your solution, I am still not seeing flagged message in parent report

Well I would have to recheck since my proposal is a week old.

**Comment by Talha345** [comment]
@aimane-chnaif the codebase has been updated in the meantime so a few changes would have to be made to my original proposal but it is still valid and I see the following in the parent report after flagging a message.

<img width="1494" alt="image" src="https://github.com/Expensify/App/assets/36226639/7502f94a-a62b-4189-b2b9-24b3bdb1f8d9">

**Comment by Talha345** [comment]
> I agree with @tienifr `isDeletedParentAction` should be set to true when message is red flagged because we now treat this same as delete message.
> 
> We should consider many things while fixing this issue:
> 
> * title and last message text in LHN
> * report name in header
> * report action item style
> * report accessibility
> * context menu actions
> 
> Without `isDeletedParentAction` change, this issue can still be fixed in frontend only but I think it's ideal to change `isDeletedParentAction` both in frontend & backend and requires less code change.
> 
> 🎀 👀 🎀 for engineer's thoughts on backend fix

@aimane-chnaif I agree with you that we should clarify the scope of this ticket and expected behaviour before moving forward. One thing I would also like to clarify is that if a message is red flagged, we are currently setting it to `pendingRemove` and it is not deleted actually because the Concierge has to review the report so now I wonder if we should really set `isDeletedParentAction` to true on the BE.

**Comment by Talha345** [comment]
> > it is not deleted actually because the Concierge has to review the report
> 
> We're not actually deleting message, keep html and text. This is different from delete message flow. `isDeletedParentAction` flag is used when parent report action has thread, especially when thread has visible actions.

Yes I know that this flow is different from delete message and that we are not actually deleting the html and text.
So, what you are suggesting is that we show the following in the parent report in case of flagged messages also (see image below). Totally makes sense.

![Screenshot 2023-07-28 at 16 08 59](https://github.com/Expensify/App/assets/36226639/aeb062b8-25ec-4eaf-9a37-9a1a4e627201)

Once the expected behaviour is confirmed,I will update my proposal.I think rest everything is working fine with my proposal.

**Comment by Talha345** [comment]
> As per discussion, everything should visually look as if deleted.

So, even in the thread there should be no option to reveal the message, right?

**Comment by Talha345** [comment]
> Right, red flagged messages should never be revealed. That's the difference from yellow flagged messages.

Okay, got it. Will update my proposal.

**Comment by dangrous** [comment]
I have not been paying attention to this thread but just to jump in - when a message has child report actions and is pending red flagged, we're swapping it out for `[Hidden message]` to follow the pattern of `[Deleted message]` until it's actually decided to be removed by the moderators. That was a recent change in ... some PR?

I think y'all probably are aware of that but just wanted to call it out in case!

**Comment by Talha345** [comment]
> ided to be removed by the moderators. That was a recent chang

Hi @dangrous, In STG right now when a thread parent message is flagged as red, the message as well as the thread is hidden from the original chat and the thread shows a 404.

You can see that in the recording below.The thread with the parent message 'How r u' shows a 404 but the message is still shown in the LHN, however this entire thread is not anymore in the original chat.


https://github.com/Expensify/App/assets/36226639/5c50d510-320b-4650-ac82-5ba4142e50fc

**Comment by dangrous** [comment]
Hrmmmm maybe https://github.com/Expensify/App/issues/21890 doesn't work as intended? Or we are handling it slightly differently in different contexts?

Yeah I think in that Slack thread you shared @aimane-chnaif I was thinking only of messages that have actually been removed (vs. `pendingRemove`) by a moderator. I think this is the full expected behavior:
* Message removed by moderator - `[Deleted message]` if parent, totally gone if not (mirrors a regularly deleted message)
* Message reported with `pendingRemove` status - `[Hidden message]` in LHN and thread header, "This message has been removed..." in chat window.

Does that make sense?

**Comment by Talha345** [comment]
> What is status after message removed by moderator?

No idea.Maybe @dangrous can tell us!

**Comment by Talha345** [comment]
> * header

@dangrous What about if message is in `pendingRemove`, do we show it in the original report?

**Comment by dangrous** [comment]
so if it's removed by the moderator it'll effectively be fully deleted - technically there's a decision of `removed` but since the text of the message is set to `''` it will just never show up (treated just as if it's deleted by a user)

**Comment by dangrous** [comment]
if pending, if no children it'll be invisible in the original report, if there are children it will be `[Hidden message]`

**Comment by Talha345** [comment]
> if pending, if no children it'll be invisible in the original report, if there are children it will be `[Hidden message]`. It will also be [Hidden message] in the LHN and thread headers.
> 
> ^^ this is correct, what I stated [above](https://github.com/Expensify/App/issues/22383#issuecomment-1655822498) is incorrect.

@dangrous Why do we have different behaviour in case if the message has children vs no children.Shouldn't we just aim for consistent behaviour and just show [Hidden Message].Also do we have to show the option to reveal the message in this case in the original report and thread report or just simply display [Hidden Message].

Also I do not see any decision status `removed` in `CONST.js` file, but does that mean if the moderator approves the flagged message as harrasment/bullying, it is automatically set as `''`.

**Comment by Talha345** [comment]
@dangrous @aimane-chnaif Just to make sure, everyone is on the same page, I am writing the expected behaviour below:

1. If a message is flagged as red and is in status `pendingRemove`:

-  In Original Report:
    - If a message has children --> [Hidden Message]
    - If a message does not have children --> Invisible
    
- In Thread Report, LHN and Thread Header --> [Hidden Message]

2. When the message is marked as removed by the moderator, it will be set as `''` from the backend.

-  In Original Report:
    - If a message has children --> [Deleted Message]
    - If a message does not have children --> Invisible
    
 - In Thread Report, LHN and Thread Header --> [Deleted Message]

Open Questions: 

1. In the first scenario do we simply show [Hidden Message] or 'This message has been flagged as violating our community rules and the content has been hidden' with the 'Reveal message' button like shown below:

![image](https://github.com/Expensify/App/assets/36226639/ac4303a1-4ef4-4697-aceb-57c68e9d1767)

2. In the first scenario, shouldn't we just aim for consistent behaviour and show whatever we decide in the Q above for both cases i.e a message has children or not?

**Comment by Talha345** [comment]
@dangrous @aimane-chnaif Any updates on my [comment](https://github.com/Expensify/App/issues/22383#issuecomment-1656171988) above so we can proceed this task further?

**Comment by dangrous** [comment]
I think you are correct on the expected behavior! As for your questions:
1. This was me being confusing. Sorry! If the message is "red" flagged, we should show `[Hidden message]` **NOT** `This message has been flagged as violating our community rules and the content has been hidden`. That (with the button) should only be shown for "yellow" flags.
2. Overall, we're only showing `[Deleted message]` or `[Hidden message]` when we *have* to. Otherwise they're visually hidden.

**Comment by Talha345** [comment]
@dangrous Thanks for the clarification.Last question 😅 : How can we test the flow of getting the message removed from the moderator after it has been flagged? Would you or some other Internal Engineer be available for testing this workflow?

**Comment by Talha345** [comment]
> So yeah, two ways - either I or another engineer can sync with you and provide a link to use in our backend
> 
> or
> 
> As long as it's not too many, you can just flag messages that say something like "Hi there! Can you please remove **[OR WHATEVER YOU NEED THEM TO DO]** this message? I'm a contributor testing an improvement to moderation."

@dangrous I think syncing with an engineer will make more sense because I do not know how much time the moderators will take to remove the flagged messages!

**Comment by dangrous** [comment]
Yep! That probably makes the most sense

**Comment by Talha345** [comment]
@aimane-chnaif @dangrous Since there were quite some changes needed to fix this issue and everything could not be presented in a proposal, I went ahead and created a PR which fixes this issue.You can view the changes here https://github.com/Expensify/App/pull/23960 .It addresses all the points mentioned in the first scenario [here](https://github.com/Expensify/App/issues/22383#issuecomment-1656171988). 

Regarding the second scenario i.e the message is marked as removed by the moderator, as @dangrous pointed out that the message will be deleted so I believe it will follow the deleted message flow and we would not need to make any further changes but that still needs to be tested with an internal engineer as per our discussion above.

**Comment by Talha345** [comment]
> @Talha345 your proposal is not approved yet. If you wanted me to test, you could share your local branch. Close that PR.

@aimane-chnaif PR closed, I just created the PR in order to make it easier for you to view the changes.You can test the changes from the branch `Talha345:fix/22383`

**Comment by johncschuster** [comment]
@aimane-chnaif can you take a look at @tienifr's [proposal](https://github.com/Expensify/App/issues/22383#issuecomment-1624845324) above? Is it sufficient? If not, can you make this issue your highest priority?

**Comment by Talha345** [comment]
> @aimane-chnaif can you take a look at @tienifr's [proposal](https://github.com/Expensify/App/issues/22383#issuecomment-1624845324) above? Is it sufficient? If not, can you make this issue your highest priority?

@johncschuster I already shared a working proposal [here](https://github.com/Expensify/App/issues/22383#issuecomment-1659145901) and an associated branch with it (see [here](https://github.com/Expensify/App/issues/22383#issuecomment-1659153093)) for @aimane-chnaif to test since the changes were quite some and I thought sharing a working branch would be better.

Also, my proposal with the branch came first and @tienifr updated his proposal after I presented my branch and PR and his changes are almost identical to mine.Also if you read the entire conversation in this issue,I spent a lot of time and effort in order to communicate with C+ and internal engineer to clarify the scope and the expected behaviour of this ticket.

Not sure how you missed all this.😅

**Comment by Talha345** [comment]
@aimane-chnaif Any update on the new proposals? Also I have updated my original proposal https://github.com/Expensify/App/issues/22383#issuecomment-1638723258 with the same changes that I had already proposed in my branch and PR in these comments: https://github.com/Expensify/App/issues/22383#issuecomment-1659145901 , https://github.com/Expensify/App/issues/22383#issuecomment-1659153093 in order to make the changes clear for anyone who views this issue and does not read the entire conversation.

**Comment by Beamanator** [comment]
@dangrous can you take over this issue since you're way involved in this feature & convo? 🙏

**Comment by dangrous** [comment]
Sure! Looks like there's no current action needed from me, though, right? I'll unassign you and add me!

**Comment by Talha345** [comment]
> Sure! Looks like there's no current action needed from me, though, right? I'll unassign you and add me!

Yes, still waiting on @aimane-chnaif to approve a proposal.

**Comment by Talha345** [comment]
@aimane-chnaif Please find below attached the demo videos:

Online:

https://github.com/Expensify/App/assets/36226639/2330448a-3e26-479a-a4dd-8360fff64675

Offline:


https://github.com/Expensify/App/assets/36226639/3aeaa3a7-157e-4b00-9265-fadb1c93f015

**Comment by Talha345** [comment]
@aimane-chnaif Any update on this?
I have already shared a working proposal, branch and result videos after the fix!

**Comment by Talha345** [comment]
@aimane-chnaif  @johncschuster Bump!

**Comment by dangrous** [comment]
how are we looking @aimane-chnaif? Do we need to look for other proposals, or is @Talha345's approach looking good? (I haven't had a chance to review yet)

**Comment by Talha345** [comment]
@dangrous I have also shared a working branch along with my proposal and have updated it multiple times with main because this issue has now been open for 6 weeks and the code base is being updated regularly.

I thought there was a 4 weeks time limit until an issue breaks WAQ. Also what is the policy if the C+ member is not active because you can see @aimane-chnaif posted 5 days ago that he will approve today but still hasn't.I guess he is overburdened with work so in this scenario shouldn't another C+ be assigned?

**Comment by dangrous** [comment]
Hey @aimane-chnaif how are we looking, do we need to reassign this to another C+ if you're too busy?

**Comment by situchan** [comment]
I will take this

**Comment by johncschuster** [comment]
Sweet. Thanks, @situchan!

**Comment by Talha345** [comment]
Hi @situchan, Just to give you a recap of this issue, the expected behaviour was decided here: https://github.com/Expensify/App/issues/22383#issuecomment-1656171988. I have provided a proposal for it here https://github.com/Expensify/App/issues/22383#issuecomment-1638723258 as well as the mentioned changes in a draft PR here https://github.com/Expensify/App/pull/23960, since there were quite some changes to be done at different places in the codebase.I have also shared the videos demonstrating the expected behaviour after applying my changes here https://github.com/Expensify/App/issues/22383#issuecomment-1674886319.

Do let me know if you have any questions!

**Comment by dangrous** [comment]
Thanks for taking over @situchan!

**Comment by Talha345** [comment]
@situchan Bump please as this issue is pretty old now and I think quite important too!

**Comment by situchan** [comment]
So far I reviewed all discussions and confirmed the expected behavior.
Reviewing 2 proposals are on my schedule today

**Comment by Talha345** [comment]
> So far I reviewed all discussions and confirmed the expected behaviour.
> Reviewing 2 proposals are on my schedule today

@situchan Great! Looking forward to an update! I have updated the draft PR https://github.com/Expensify/App/pull/23960 (containing my proposal) again with main so you can test it also if required.I have also provided a summary of this issue here https://github.com/Expensify/App/issues/22383#issuecomment-1694355574

**Comment by Talha345** [comment]
Hi @situchan! Any update?

**Comment by johncschuster** [comment]
Bumping the issue to make Melvin happy. Any update on this one, @situchan?

**Comment by situchan** [comment]
I will provide update soon. Also answered in https://expensify.slack.com/archives/C02NK2DQWUX/p1693927748748159?thread_ts=1692804121.042219&cid=C02NK2DQWUX

**Comment by dangrous** [comment]
Hey @situchan - can we get an update by the end of this week? Thanks!

**Comment by dangrous** [comment]
Following up @situchan - let us know if we need to reassign this issue

**Comment by situchan** [comment]
updating today as merge freeze is over

**Comment by situchan** [comment]
@Talha345 I don't think your [proposal](https://github.com/Expensify/App/issues/22383#issuecomment-1638723258) is handling LHN last message. Can you please update your proposal fixing that as well?
<img width="1152" alt="Screenshot 2023-09-12" src="https://github.com/Expensify/App/assets/108292595/7e2b15f7-d3c8-4d1a-b958-38cabb94305e">

**Comment by situchan** [comment]
@Talha345 above screenshot is from your draft [PR](https://github.com/Expensify/App/pull/23960)

**Comment by Talha345** [comment]
> @Talha345 above screenshot is from your draft [PR](https://github.com/Expensify/App/pull/23960)

@situchan okay let me check, can you kindly let me know the steps to reproduce as there are multiple scenarios in this task!

**Comment by situchan** [comment]
@Talha345 

- when no child message in thread

https://github.com/Expensify/App/assets/108292595/33afe242-3012-40c7-98f2-4924eea2301b



- when child message exists in thread

https://github.com/Expensify/App/assets/108292595/1db99044-e61d-491f-b2d9-d3afd101751c

**Comment by Talha345** [comment]
> @Talha345
> 
> * when no child message in thread
> 
>  Screen.Recording.2023-09-12.at.6.15.06.PM.mov 
> * when child message exists in thread
> 
>  Screen.Recording.2023-09-12.at.6.16.02.PM.mov

Okay I am checking this and will update in a while!

**Comment by Talha345** [comment]
@situchan I have fixed the LHN last message issue, you can see the change in my commit https://github.com/Expensify/App/pull/23960/commits/cfed4bbf6e0237c5943b9c0788c95c6c6ba48ff9. Also made the update in my proposal https://github.com/Expensify/App/issues/22383#issuecomment-1638723258

**Comment by situchan** [comment]
> @situchan I have fixed the LHN last message issue, you can see the change in my commit [cfed4bb](https://github.com/Expensify/App/commit/cfed4bbf6e0237c5943b9c0788c95c6c6ba48ff9). Also made the update in my proposal [#22383 (comment)](https://github.com/Expensify/App/issues/22383#issuecomment-1638723258)

Still not working for me

**Comment by Talha345** [comment]
@situchan It seems to be working fine for me.See attached video below:


https://github.com/Expensify/App/assets/36226639/7685fd34-e3a2-4a26-8dee-200b117cd135

**Comment by situchan** [comment]
yes, working fine. I forgot the behavior that [Hidden message] should not be present at all in LHN

**Comment by situchan** [comment]
Here's my conclusion:
@tienifr and @Talha345 contributed together on this issue.
While @Talha345 followed discussion and defined the expected behavior, I like @tienifr's implementation better, though their proposals are similar. i.e. we should implement [Hidden message] similar to [Deleted message] in ReportActionItemFragment. Otherwise, it will break text style and there will be inconsistency.
And @tienifr updated their proposal first based on the expected behavior (as well as before requirement change), while @Talha345 tried to explain in their branch code, not in proposal.
So this is what I suggest:
Assign both and split compensation. @tienifr raise PR and @Talha345 review & test.
Please let me know what you guys think.

**Comment by Talha345** [comment]
@situchan I do not agree with the statement that my and @tienifr's contributions are equal in this issue. I am sure you have followed the entire conversation above and seen that 90% of the comments would be mine because I spent a lot of time and effort to clarify the scope and the expected behaviour of this issue while @tienifr just updated his/her proposal when I was done clarifying the expected behaviour.

The reason why I presented my changes in a branch first and then later on in the formal proposal was because I was having a conversation with the previous C+ and I thought it was better to provide a working branch/draft PR so that the C+ can view the code changes since they were scattered in different files and proposing them directly in the proposal initially did not make much sense (I am sure you would have felt that if you just read the proposal without the draft PR).Nevertheless, my changes in the branch still came first before @tienifr update his/her proposal.

Lastly, I agree that both our proposals are quite similar apart from the fact that @tienifr's proposal implements [Hidden message] in `ReportActionItemFragment` whereas I do that in `ReportActionItemMessage`, however there is a reasoning for that and that is that the `moderationDecision` is not present in `ReportActionItemFragment` and would have to be passed from `ReportActionItem` to `ReportActionItemFragment` via `ReportActionItemMessage`. @tienifr's proposal does not even explain this and therefore the proposal is not in a working state. Apart from that, the codebase has changed a lot in the meantime and I have spent time and effort to maintain and update the draft PR with the latest main and you can see that even the draft PR has deviated a bit from my latest proposal.This was the core reason that I presented my changes in a draft PR too since the codebase changes quite fast and one would have to update their proposal after a few days since this issue is now more than 2 months old. Furthermore,@tienifr's proposal also does not cater the latest changes in the codebase and the LHN message that I just updated, and therefore IMO it is incomplete.

**Comment by situchan** [comment]
> @tienifr's proposal does not even explain this and therefore the proposal is not in a working state

This is not mandatory in proposal stage. Neither is draft PR.

**Comment by Talha345** [comment]
> @Talha345 what's your plan to fix `[Hidden message]` inconsistency?

IMO it makes sense to display `[Hidden message]` in `ReportActionItem` since we do not need to go two components deep by passing extra arguments and execute arbitrary code when we can achieve the same using an `if-else` clause which will also provide performance advantage. It is important to keep in mind that switching components results in extra overhead of context switching.

On the other hand, I understand what you mean by inconsistency and want to maintain separation of concerns and the obvious solution to do that would be to pass the moderationDecision via props to children components and implement the logic in `ReportActionItemFragment`.

However, this point is a minor implementation difference and should not be the deciding factor in proposal selection IMO.

**Comment by situchan** [comment]
I won't say "creating branch" is earlier than "proposal updated".
I understand the efforts you made on this GH so this is edge case.
But for next issues, don't create branch but use plain English in proposal like what @tienifr did.

@dangrous will give the final call. 2 options:
- assign only @Talha345 
- assign both and split comp

**Comment by dangrous** [comment]
Hi! Apologies, I was traveling. I will have an update shortly!

**Comment by Talha345** [comment]
Hi @dangrous! Do you have any update yet?

**Comment by dangrous** [comment]
Hi all! Thanks so much for your patience while we sorted out the best way to move forward here. The contribution guidelines state that we look for the “earliest provided, best” proposals, and in this case, @tienifr provided the proposal that matches those guidelines. We acknowledge that there was a lot of work by multiple parties in this issue, but it’s important to understand that we try to stick to the contribution rules at all times, to maintain a fair chance for all contributors. However, for the future, we will be starting a discussion to discuss whether we should officially allow for test branches to count as proposals in the future, and we'll update in expensify-open-source when that's settled. For this PR, we will move forward with assigning @tienifr but please keep looking for more issues you can contribute to, and get those proposals submitted!

**Comment by Talha345** [comment]
@dangrous Even though @tienifr's proposal was updated earlier, it is incomplete and not even updated with the latest codebase and therefore not valid completely not and I have mentioned that in my previous comments.
Honestly, I am pretty disappointed to see that being proactive and figuring out the best way forward for an issue (which is an essential part of software engineering) is ignored and the only thing that is being considered is who updated the proposal first.

Nevertheless, I would say lesson learnt for sure and best of luck to @tienifr for this issue.

**Comment by dangrous** [comment]
Yep definitely understand where you're coming from @Talha345 - we internally confirmed that we have to stick with the letter of the law/guidelines and only look at posted proposals, which leads to this outcome for this issue. Your work has not gone unappreciated or unnoted, though, and we'll be excited to work with you on other issues in the future! Just make sure to post in proposal form moving forward.

**Comment by situchan** [comment]
I raised discussion on what to show as last message in LHN - https://expensify.slack.com/archives/C01GTK53T8Q/p1695604109450219.

Let's hold this until we get confirmation about the expected behavior.

<img width="1788" alt="bug?" src="https://github.com/Expensify/App/assets/108292595/37058357-d8c9-4f82-be7c-38d505b0d8ff">

**Comment by dangrous** [comment]
PR is in process but in hold for https://github.com/Expensify/App/pull/28364 because it affects similar pieces in the LHN and is needed to confirm testing

**Comment by situchan** [comment]
Maybe we were lastly blocked on [this bug](https://github.com/Expensify/App/pull/28002#issuecomment-1772318137) from backend.
@tienifr can you pull main and retest? If still happens, please answer on https://github.com/Expensify/App/pull/28002#issuecomment-1773092463

**Comment by situchan** [comment]
@tienifr bump

**Comment by dangrous** [comment]
Two PRs are in process to fix the childType issue that's holding up this PR

**Comment by dangrous** [comment]
PRs are merged but not yet deployed - @tienifr and @situchan we should be able to move forward with testing this PR shortly

**Comment by dangrous** [comment]
Update, this has been a challenge to get tested and reviewed due to many many merge conflicts. We're (@situchan and @tienifr ) are pushing on this now - do we think we can get it merged by the end of the week?

If we need to coordinate a specific time to merge and then review immediately after, maybe that would be best?

**Comment by dangrous** [comment]
We're so close! One final edit and then I think this is ready to push through

**Comment by johncschuster** [comment]
Congrats on getting this one to the finish line! I'll get payment issued after the 7-day window.

**Comment by johncschuster** [comment]
Payment has been issued! 🎉 

@situchan, can you complete the BZ Checklist above when you get a moment? Thank you!

**Comment by situchan** [comment]
This was very old issue and not able to find offending PR. At least not a recent regression.

### Regression Test Proposal
1. Send a message from user A to user B
2. From the tab where user B is signed-in, reply to A's message in a thread
3. Flag the parent message in the thread (the message you replied to) as Harassment or Assault
4. Verify that the "Hidden" message is shown in parent chat
5. Click on the thread chat, verify that the "Hidden" message is shown in the header and in the first action

**Comment by Natnael-Guchima** [comment]
Hello @johncschuster, this issue was reported before August 30 when the bug bounty was 250. It seems that you have sent me 50. This is to ask if this is something that can be fixed. And no worries if it doesn't have to be fixed. 
Thanks!

**Comment by johncschuster** [comment]
@Natnael-Guchima oh! That was my oversight. I'll get that corrected. Thanks for calling it out!

**Comment by johncschuster** [comment]
I've just issued an additional $200 to you @Natnael-Guchima

**Comment by johncschuster** [comment]
I'll get the Regression test issue going on Monday

**Comment by Natnael-Guchima** [comment]
Received. Thanks @johncschuster 🙏
